packages/x-components/src/x-modules/search/store/actions/batch-state-resets-after-request-update.action.ts

```diff
@@ -0,0 +1,63 @@
+import { isArrayEmpty } from '../../../../utils/array';
+import { InternalSearchRequest } from '../../types';
+import { SearchXStoreModule } from '../types';
+
+const baseKeys: (keyof InternalSearchRequest)[] = [
+  'filters',
+  'page',
+  'query',
+  'relatedTags',
+  'selectedFilters',
+  'sort'
+];
+
+/**
+ * Default implementation for the {@link SearchActions.batchStateResetsAfterRequestUpdate}.
+ *
+ * @param context - The {@link https://vuex.vuejs.org/guide/actions.html | context} of the actions,
+ * provided by Vuex.
+ *
+ * @public
+ */
+// eslint-disable-next-line max-len
+export const batchStateResetsAfterRequestUpdate: SearchXStoreModule['actions']['batchStateResetsAfterRequestUpdate'] =
+  ({ commit }, { newRequest, oldRequest }) => {
+    const keys = getKeysWithDifferentValue<InternalSearchRequest>(newRequest, oldRequest);
+    const hasExtraParamsKeys = keys.some(key => !baseKeys.includes(key));
+
+    if (!isArrayEmpty(keys)) {
+      if (!keys.includes('page')) {
+        commit('setPage', 1);
+      }
+      if (keys.includes('query')) {
+        commit('setSort', '');
+      }
+      if (hasExtraParamsKeys) {
+        commit('setFacets', []);
```

Comment:
After talking with you about this, I still don't know why it is needed. I find this facets case very similar to the `relatedTags` one which aren't included here. So, should we just deselect the selected filters in the `facets` module?

Reply:
I've test that and it seems to work fine. I just ported what was being reseted before, and I agree that deselecting the facets have more sense than removing them from the search state.